app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -0,0 +1,446 @@
+import android.hardware.camera2.CameraManager;
+import android.hardware.camera2.CameraMetadata;
+import android.hardware.camera2.CaptureRequest;
+import android.hardware.camera2.TotalCaptureResult;
+import android.hardware.camera2.params.StreamConfigurationMap;
+import android.media.Image;
+import android.media.ImageReader;
+import android.os.Bundle;
+import android.os.Environment;
+import android.os.Handler;
+import android.os.HandlerThread;
+import android.support.annotation.NonNull;
+import android.support.v4.app.ActivityCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.util.Size;
+import android.util.SparseIntArray;
+import android.view.Surface;
+import android.view.TextureView;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.List;
+
+/**
+ * The Activity useful for making photos
+ */
+public class CameraActivity extends AppCompatActivity {
+
+    // Final shared variables
+    /**
+     * TAG used for logs
+     */
+    private static final String TAG = "CameraActivity";
+    /**
+     * directory used to save images and/or text recognized
+     */
+    private static final File SAVE_DIR = new File(Environment.getExternalStorageDirectory().getAbsolutePath() + "/OCRCamera/Pictures/");
+
+    //creation intent
+
+    /**
+     * Code for the permission requested
+     */
+    private final int MY_CAMERA_REQUEST_CODE = 200;
+
+    /**
+     * SparseIntArray used for the conversion from screen rotation to Bitmap orientation
+     */
+    private static final SparseIntArray ORIENTATIONS = new SparseIntArray();
+    static{
+        ORIENTATIONS.append(Surface.ROTATION_0,90);
+        ORIENTATIONS.append(Surface.ROTATION_90,0);
+        ORIENTATIONS.append(Surface.ROTATION_180,270);
+        ORIENTATIONS.append(Surface.ROTATION_270,180);
+    }
+
+    // Shared variables
+    /**
+     * the texture view used for show the preview of the camera
+     */
+    private TextureView mCameraTextureView;
+    /**
+     * the string where i save the camera id
+     */
+    private String mCameraId;
+    private Handler mBackgroundHandler;
+    private HandlerThread mBackgroundThread;
+    /**
+     * The CameraDevice used to interact with the physical camera.
+     */
+    private CameraDevice mCameraDevice;
+
+    /**
+     * CaptureRequest.Builder used for the camera preview.
+     */
+    private CaptureRequest.Builder mCaptureRequestBuilder;
+
+    /**
+     * CameraPatureSession used for the camera preview;
+     */
+    private CameraCaptureSession mCameraCaptureSession;
+
+    /**
+     * Size used for the camera preview
+     */
+    private Size mPreviewSize;
```

Comment:
I think this name isn't clear enough. I would have used something like "cameraPreviewSize"